src/grpc/mod.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use futures::channel::oneshot;
+use futures::executor::block_on;
+use futures::prelude::*;
+use grpcio::{Environment, ServerBuilder};
+use std::error::Error;
+use std::io::Read;
+use std::str;
+use std::sync::Arc;
+use std::{io, thread};
+use string_error::into_err;
+
+pub mod grpc_keyprovider;
+
+use grpc_keyprovider::keyprovider;
+use grpc_keyprovider::keyprovider_grpc::*;
+use grpc_keyprovider::keyprovider_structs::*;
+
+use super::kbc_manager;
+
+#[derive(Clone)]
+pub struct KeyProvider;
+
+impl KeyProviderService for KeyProvider {
+    fn un_wrap_key(
```

Comment:
Seems like unwrap should be one word? I guess this is autogenerated somewhere?

Reply:
Yes, the name here is automatically generated by the protobuf compiler.

Reply:
I guess this is because the protobuf says UnWrapKey rather than UnwrapKey. Looks like ocirypt is a bit inconsistent in the naming. Not a big deal.